assets/src/edit-story/components/panels/textStyle/padding.js

```diff
@@ -45,14 +45,26 @@ const Space = styled.div`
   flex: 0 0 10px;
 `;
 
-function PaddingControls({ selectedElements, pushUpdateForObject }) {
+function PaddingControls({
+  selectedElements,
+  pushUpdateForObject,
+  pushUpdate,
+}) {
   const padding = useCommonObjectValue(
     selectedElements,
     'padding',
     DEFAULT_PADDING
   );
+  const isPaddingLock =
+    getCommonValue(selectedElements, 'lockPadding') === MULTIPLE_VALUE
+      ? true
+      : getCommonValue(selectedElements, 'lockPadding');
 
-  const [lockPaddingRatio, setLockPaddingRatio] = useState(true);
+  const [lockPaddingRatio, setLockPaddingRatio] = useState(isPaddingLock);
+
+  useEffect(() => {
+    setLockPaddingRatio(isPaddingLock);
+  }, [isPaddingLock]);
```

Comment:
I still think it's a bit weird we have three completely differently named variables here, that all talk about the same thing: `lockPadding`, `isPaddingLock`, and `lockPaddingRatio`. I actually don't even think we need the third one (local state) at all, and the two remaining can be named the same. Try this:

```js
const rawLockPadding = getCommonValue(selectedElements, 'lockPadding');
const lockPadding = rawLockPadding === MULTIPLE_VALUE ? true : rawLockPadding;
```

And then completely drop `lockPaddingRatio` and `setLockPaddingRatio` including the `useEffect` here - and also just [delete line 104](https://github.com/google/web-stories-wp/pull/1075/files#diff-276b8cf012e711e1e6d37e55f529ff49R104).

I think the value from the store will come back "fast enough" in order for the component to stay "controlled" - it's not debounced, right?

If you can't remove the local state, then e.g. rename the third variable to `lockPaddingLocal` (and `setLockPaddingLocal`), so it's clear it's the same thing, just locally in this component.

Reply:
Yes, we don't need third one. Updated!